### DISCUSSION TITLE
Test support for older OS versions

### DISCUSSION THREAD

**Discussion by tylervick** [MobileNativeFoundation/discussions#90] [General]
Hey everyone,

I'm curious to hear about testing support for previous OS versions. Often times developers will keep support for older versions to hold onto the long tail of non-upgrading users (particularly on Android, but iOS too).

While it's generally easy to continue distributing new releases to old OSes, I find that maintaining stable automated tests is more difficult. Whether it's broken APIs or worse performance on old hardware, justifying continued support can grow difficult over time.

What are some experiences here? Does automated testing align with officially-supported versions, or just focusing on the newer versions?

**Comment by keith** [comment]
Realistically I'd say we only test against the newest version most of the time. We only run our test suites against the newest version at this point (we used to support running UI tests against older versions, but no one really used it). QEs often have devices with older OS versions but I guess it seems like it's up to them to decide what they want to test on for normal testing.

We actually use this as a motivation to drop older OS support given that it's quite easy for us to ship a bug when something subtle differs from version to version. And realistically when you look at our user breakdowns it rarely seems worth spending much time testing on the older versions.
